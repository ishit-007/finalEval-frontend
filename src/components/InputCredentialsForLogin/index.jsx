/* eslint-disable no-unused-vars */
import './index.css';
import React from 'react';
import axios from 'axios';
// import PropTypes from 'prop-types';
import { useNavigate } from 'react-router-dom';
import { AxiosError } from 'axios';

const InputCredentialsForLogin = () => {
  const loginHandler = e => {
    axios
      .post('http://localhost:4000/login', {
        userName: userName,
        password: password,
      })
      .then(resp => {
        const token = resp.data;
        if (token) {
          localStorage.setItem('token', token);
          navigate('/content-types');
        }
      })
      .catch(err => {
        alert('Invalid Credentials');
      });
  };
  const [userName, setUserName] = React.useState('');
  const [password, setPassword] = React.useState('');
  const userNameChangeHandler = event => {
    setUserName(event.target.value);
  };
  const passwordChangeHandler = event => {
    setPassword(event.target.value);
  };
  const navigate = useNavigate();
  return (
    <div className="inputForm">
      <div className="heading">
        <h1>Login to your CMS+ Account</h1>
      </div>
      <div className="emailInput">
        <label htmlFor="">Email</label>
        <input type="email" name="email" id="email" onChange={userNameChangeHandler} />
      </div>
      <div className="passwordInput">
        <label htmlFor="">Password</label>
        <input type="password" name="password" id="password" onChange={passwordChangeHandler} />
      </div>
      <div className="submitBtn">
        <button type="button" onClick={loginHandler}>
          Login
        </button>
        <a href="">Forgot Password?</a>
        <a
          href=""
          className="other-Option"
          onClick={() => {
            navigate('/register');
          }}>
          Register Instead
        </a>
      </div>
    </div>
  );
};

export default InputCredentialsForLogin;
