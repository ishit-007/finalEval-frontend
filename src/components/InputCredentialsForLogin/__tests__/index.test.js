import InputCredentialsForLogin from '..';
import renderer from 'react-test-renderer';
import React from 'react';
jest.mock('react-router-dom');
describe('InputCredentials for Login', () => {
  it('should render the InputCredentials component', async () => {
    const tree = renderer.create(<InputCredentialsForLogin />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});