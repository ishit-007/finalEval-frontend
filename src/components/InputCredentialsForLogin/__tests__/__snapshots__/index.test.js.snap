// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputCredentials for Login should render the InputCredentials component 1`] = `
<div
  className="inputForm"
>
  <div
    className="heading"
  >
    <h1>
      Login to your CMS+ Account
    </h1>
  </div>
  <div
    className="emailInput"
  >
    <label
      htmlFor=""
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      type="email"
    />
  </div>
  <div
    className="passwordInput"
  >
    <label
      htmlFor=""
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      onChange={[Function]}
      type="password"
    />
  </div>
  <div
    className="submitBtn"
  >
    <button
      onClick={[Function]}
      type="button"
    >
      Login
    </button>
    <a
      href=""
    >
      Forgot Password?
    </a>
    <a
      className="other-Option"
      href=""
      onClick={[Function]}
    >
      Register Instead
    </a>
  </div>
</div>
`;
